src/containers/widget/index.js

```diff
@@ -65,18 +45,17 @@ class Widget extends Component {
   }
 
   componentDidMount() {
-    this.setState({
-      ...getBasicState(this.props),
-    });
+    this.setDefaults();
   }
 
   setDefaults = () => {
     this.setState({
+      ...getInitialState(this.props),
       isModalOpen: false,
       invoiceDetails: null,
       fetchingInvoiceState: false,
       currentScreen:
-        this.props.welcomeMessage === "" ? "donate-screen" : "welcome-screen",
+        !this.props.welcomeMessage || this.props.welcomeMessage === "" ? "donate-screen" : "welcome-screen",
```

Comment:
Just a nitpic: if we set `welcomeMessage: ""` in `defaultProps`, we can reduce the double check here.